compat/src/index.d.ts

```diff
@@ -30,6 +30,7 @@ declare namespace React {
 	export import useReducer = _hooks.useReducer;
 	export import useRef = _hooks.useRef;
 	export import useState = _hooks.useState;
+	export type Dispatch<A> = (value: A) => void;
```

Comment:
```suggestion
	export type Dispatch<A> = (action: A) => void;
```

Might be worth extracting to core's types for reuse in submodules, not sure. We do use it in a few different places.

Reply:
@rschristian Do you mean define the type in `./src/index.d.ts` as part of preact's namespace and re-export in the compat types via `./compat/src/index.d.ts`?

Reply:
Sorry, not core, but to hooks, and importing from there, as it could directly be used in [`useReducer`](https://github.com/preactjs/preact/blob/bd593c49423f1f49a4cfbd46bbe53f8a396aaaa1/hooks/src/index.d.ts#L30)'s types. 

Apologies, was looking at a couple of different issues at once and mixed them up.

Reply:
Okay sounds good. I'm making this change now then will mark this as ready for review.